src/Components/CreateImageWizard/steps/Packages/Packages.tsx

```diff
@@ -793,12 +810,23 @@ const Packages = () => {
         setIsSelectingPackage(pkg);
       } else {
         dispatch(addPackage(pkg));
+        if (pkg.sources && pkg.sources[0].type === 'module') {
+          dispatch(
+            addModule({
+              name: pkg.sources[0].name || '',
+              stream: pkg.sources[0].stream || '',
+            })
+          );
+        }
         setCurrentlyRemovedPackages((prev) =>
           prev.filter((curr) => curr.name !== pkg.name)
         );
       }
     } else {
       dispatch(removePackage(pkg.name));
+      if (pkg.sources && pkg.sources[0].type === 'module') {
+        dispatch(removeModule(pkg.name));
+      }
```

Comment:
Doesn't this need to check if this was the only package with that module as a package source? What if there were two packages from the same module stream selected?

Also I tihnk `removeModule(pkg.name)` needs to be `removeModule(pkg.source[0].name`

Reply:
Great point, fixed!

>Also I tihnk removeModule(pkg.name) needs to be removeModule(pkg.source[0].name

Oof, overlooked that one, fixed as well, thank you!

Reply:
Wait
```

      if (
        pkg.sources &&
        pkg.sources[0].type === 'module' &&
        pkg.sources[0].name
      ) {
        dispatch(removeModule(pkg.sources[0].name));
      }
```

now removes the module, but this will still remove a module even if there's another package selected belonging to the same module, no?

Reply:
I've added the check to the `removeModule` action in `wizardSlice`, this should be diff of the changes: https://github.com/osbuild/image-builder-frontend/compare/22de590f49604642fe8be637969dcd78de423d4f..16e5dcae45dab6b946ab2b077336fffe1cb0c846